ToDoList-Redux/Binders/ReduxFramework.swift

```diff
@@ -83,47 +101,28 @@ extension World {
     )
 }
 
-
-// MARK: - PRISM
-
-extension AppAction {
-    public var task: TaskList.Action? {
-        get {
-            guard case let .task(value) = self else { return nil }
-            return value
-        }
-        set {
-            guard case .task = self, let newValue = newValue else { return }
-            self = .task(newValue)
-        }
-    }
-}
-
-
 // MARK: - EXTENSIONS
 
 extension TaskList {
-    
     static func viewModel<S: StoreType>(store: S) -> ObservableViewModel<TaskList.Action, TaskList.State>
-    where S.ActionType == AppAction, S.StateType == AppState {
+    where S.ActionType == TaskAction, S.StateType == [Task] {
         store
             .projection(action: Self.transform, state: Self.transform)
             .asObservableViewModel(initialState: .empty)
     }
     
-    private static func transform(_ viewAction: TaskList.Action) -> AppAction? {
+    private static func transform(_ viewAction: TaskList.Action) -> TaskAction? {
         switch viewAction {
-            case let .add(title): return .task(.add(title))
-            case let .remove(offset): return .task(.remove(offset))
-            case let .move(offset, index): return .task(.move(offset, index))
-            case let .toggle(id): return .task(.toggle(id))
+            case let .add(title): return .add(title)
+            case let .remove(offset): return .remove(offset)
+            case let .move(offset, index): return .move(offset, index)
         }
     }
     
-    private static func transform(from state: AppState) -> TaskList.State {
+    private static func transform(from state: [Task]) -> TaskList.State {
         TaskList.State(
-            title: state.task.title,
-            tasks: state.task.tasks
+            title: "Tasks",
+            tasks: state.map { State.Item(id: $0.id) }
```

Comment:
All our rows need now is id. It could even be an array of String, but because String is not identifiable you would need to use id: \.self. I personally like this more.